### PR TITLE
Thresholds/v20 backoff/v1

### PR DESCRIPTION
**Pull request by victorjulien**
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1939

#11358 plus backoff support

https://redmine.openinfosecfoundation.org/issues/7120

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21193

**Review by jasonish — CHANGES_REQUESTED** [review]
Nice, I like alot. Tested live and it looks good there. Still need a closer review. But throwing the requested changes for some doc on this new option.

File: src/detect-engine-threshold.c [hunk 1]
```diff
@@ -315,275 +642,271 @@ static inline void RateFilterSetAction(Packet *p, PacketAlert *pa, uint8_t new_a
...
+        case TYPE_RATE:
+            return 1;
+        case TYPE_THRESHOLD:
+        case TYPE_BOTH:
+            if (td->count == 1)
+                return 1;
+            return 0;
+        case TYPE_BACKOFF:
+            if (td->count == 1) {
+                te->backoff.next_value *= td->multiplier;
```
Comment (victorjulien) [inline src/detect-engine-threshold.c:683]: hah, forgot to implement overflow handling

**Comment by victorjulien** [comment]
comments addressed in #11388

**Pull request closed without merging**